### PR TITLE
Add getExperienceCloudId API + Unit tests + Integration tests

### PR DESCRIPTION
**Pull request by addb**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/AEPSDK.brs [hunk 1]
```diff
@@ -275,6 +275,45 @@ function AdobeAEPSDKInit(taskNode = invalid as dynamic) as object
...
+
+        getExperienceCloudId: function(callback = _adb_defaultCallback as function, context = invalid as dynamic) as void
+            _adb_logDebug("API: getExperienceCloudId()")
+            if callback = invalid
+                _adb_logError("setExperienceCloudId() - Cannot get ECID, invalid callback passed.")
+                return
+            end if
+
+            if context = invalid
+                _adb_logError("setExperienceCloudId() - Cannot get ECID, invalid context passed.")
```
Comment (yangyansong-adbe) [inline code/AEPSDK.brs:296]: The `context` parameter is optional, can we change it to a debug log?

File: code/main/task/eventProcessor.brs [hunk 2]
```diff
@@ -194,6 +195,16 @@ function _adb_EventProcessor(task as object) as object
             end if
         end function,
 
+        _getECID: function(event as object) as void
+            ecid = m._identityModule.getECID()
+
+            _adb_logDebug("EventProcessor::_getECID() - Dispatching getECID response event with ECID: (" + ecid + ")")
```
Comment (yangyansong-adbe) [inline code/main/task/eventProcessor.brs:201]: The retrieved `ecid` could be invalid, should we use `FormatJson(ecid)` instead?

File: code/AEPSDK.brs (hunk 1 above)
Comment (addb) [reply code/AEPSDK.brs:296]: I removed the check and the log since its optional

**Review by yangyansong-adbe — APPROVED** [review]

File: code/tests/test_public_APIs.brs [hunk 3]
```diff
@@ -351,6 +351,76 @@ sub TC_APIs_setExperienceCloudId()
...
+    context = {
+        content: "test"
+    }
+    callback_result = "testECID"
+
+    sdkInstance.getExperienceCloudId(sub(ctx, result)
+        throw "should not be called"
+    end sub, context)
+
+    sleep(5001)
```
Comment (praveek) [inline code/tests/test_public_APIs.brs:399]: Can you remove this sleep? I don't think this is needed.

File: code/tests/test_public_APIs.brs [hunk 4]
```diff
@@ -351,6 +351,76 @@ sub TC_APIs_setExperienceCloudId()
...
+    callbackInfo = sdkInstance._private.cachedCallbackInfo[event.uuid]
+
+    UTF_assertEqual(event.apiName, _internal_const.PUBLIC_API.GET_EXPERIENCE_CLOUD_ID)
+    UTF_assertInvalid(event.data)
+    UTF_AssertNotInvalid(event.uuid)
+    UTF_AssertNotInvalid(event.timestamp)
+    UTF_assertEqual(callbackInfo.context, context)
+    UTF_AssertNotInvalid(callbackInfo.timestampInMillis)
+
+    sleep(5001)
```
Comment (praveek) [inline code/tests/test_public_APIs.brs:411]: Is this the behaviour across all our APIs? Don't we signal timeout?

File: sample/development/components/integration/test/integrationTests.brs [hunk 5]
```diff
@@ -138,6 +138,105 @@ function TS_SDK_integration() as object
             return validator
         end function,
 
+        TC_SDK_getECID: function() as dynamic
```
Comment (praveek) [inline sample/development/components/integration/test/integrationTests.brs:141]: Can you add a test where getExperienceCloudId is called after setexperienceCloudID?

File: code/tests/test_public_APIs.brs (hunk 3 above)
Comment (addb) [reply code/tests/test_public_APIs.brs:399]: This is just mock delayed response to trigger timeout for callback to be called.

File: code/tests/test_public_APIs.brs (hunk 4 above)
Comment (addb) [reply code/tests/test_public_APIs.brs:411]: Yes, currently we don't error for timeout. If we want to do that, I think we will have to loop through all the callbacks and check for timeouts and error out and delete.

**Review by praveek — APPROVED** [review]

**Pull request merged**